pynats/client.py

```diff
@@ -279,7 +280,11 @@ def _readline(self, *, size: int = None) -> bytes:
         read = io.BytesIO()
 
         while True:
-            line = cast(bytes, self._socket_file.readline())
+            raw_bytes = self._socket_file.readline()
+            if not raw_bytes:
+                return None
```

Comment:
The current signature of `_readline` function does not allow to return `None`. If you want to make such changes, then you need to update return type to `Optional[bytes]` and check all dependent code.

But I think such changes are out of scope.

Reply:
Ah, yes i missed that, thanks!
Maybe cleaner solution would be to raise an Exception here instead?